server/server.js

```diff
@@ -1,15 +1,370 @@
+//     url: "https://www.youtube.com/embed/RzWB5jL5RX0",
+//     rating: 111,
+//   },
+//   {
+//     id: 442452,
+//     title: "Coding Adventure: Chess AI",
+//     url: "https://www.youtube.com/embed/U4ogK0MIzqk",
+//     rating: 671,
+//   },
+//   {
+//     id: 536363,
+//     title: "Coding Adventure: Ant and Slime Simulations",
+//     url: "https://www.youtube.com/embed/X-iSQQgOd1A",
+//     rating: 76,
+//   },
+//   {
+//     id: 323445,
+//     title: "Why the Tour de France is so brutal",
+//     url: "https://www.youtube.com/embed/ZacOS8NBK6U",
+//     rating: 73,
+//   },
+// ];
+
+// // GET "/"
+// app.get("/", (req, res) => {
+//   res.send(videos);
+// });
+
+// //POST "/"
+// app.post("/", function (req, res) {
+//   const { title, url } = req.body;
+
+//   if (!title || !url) {
+//     return res.status(400).json({
+//       result: "failure",
+//       message: "Video could not be saved",
+//     });
+//   } else {
+//     const newVideoId = videos.length + 1;
+//     const newVideo = {
+//       id: newVideoId,
+//       title,
+//       url,
+//       rating: 0,
+//     };
+//     // Adding the new video to the videos array
+//     videos.push(newVideo);
+//     res
+//       .status(201)
+//       .send(`The new video with the id ${newVideo.id} has been added`);
+//   }
+// });
+
+// // GET "/{id}"
+// app.get("/:id", function (req, res) {
+//   const videoId = Number(req.params.id);
+//   const searchedVideo = videos.find((video) => video.id === videoId);
+
+//   if (searchedVideo) {
+//     res.send(searchedVideo);
+//   } else {
+//     res.status(404).json({
+//       result: "failure",
+//       message: `Video with ID ${videoId} not found`,
+//     });
+//   }
+// });
+
+// // DELETE "/:id"
+// app.delete("/:id", function (req, res) {
+//   const videoId = Number(req.params.id);
+//   const initialVideosLength = videos.length;
+//   videos = videos.filter((video) => video.id !== videoId);
+
+//   if (videos.length < initialVideosLength) {
+//     res.send(`Video with ID ${videoId} deleted successfully.`);
+//   } else {
+//     const failureMessage = {
+//       result: "failure",
+//       message: "Video could not be deleted",
+//     };
+//     res.status(404).json(failureMessage);
+//   }
+// });
+
+// // PUT "/:id/rating"
+// app.put("/:id/rating", function (req, res) {
+//   const videoId = Number(req.params.id);
+//   const rating = req.body.rating;
+
+//   if (rating !== 1 && rating !== -1) {
+//     return res.status(400).json({
+//       result: "failure",
+//       message: "Invalid rating value. Only +1 or -1 allowed.",
+//     });
+//   }
+
+//   const video = videos.find((video) => video.id === videoId);
+
+//   if (video) {
+//     video.rating += rating;
+//     res.send(`Updated rating for the video with ID ${videoId}`);
+//   } else {
+//     res.status(404).json({
+//       result: "failure",
+//       message: `Video with ID ${videoId} not found`,
+//     });
+//   }
```

Comment:
Try and remember to delete the commented code, it makes it a bit of a mess and harder to review